### PR TITLE
Typing text

### PR DESCRIPTION
**Pull request by MarcRaeder**
add TypingText feature

### CODE REVIEW

File: my-app/src/Components/Game.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      props.setText(props.text.slice(1));
+      setWrongChar(false);
+    }
+  };
+
+  return (
+    <div className="typing-area__text">
+      <div ref={divRef} onKeyDown={onKeyPressed} tabIndex="0">
+        <>
+          {props.typedChars.map((element, i) => (
```
Comment (DevBastiS) [inline my-app/src/Components/Game.js]:
```suggestion
          {props.typedChars.map((element, index) => (
```

File: my-app/src/Components/Game.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    const correct = event.key === current && wrongChar === false;
+    const incorrect = event.key !== current;
+    const firstIncorrectThenIncorrect = current && wrongChar === true;
+
+    if (correct) {
+      const copyTypedChars = props.typedChars.slice();
+      copyTypedChars.push({
+        char: current,
+        status: "correct",
+      });
```
Comment (DevBastiS) [inline my-app/src/Components/Game.js:23]:
```suggestion
      });
      
```

File: my-app/src/Components/Game.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+      props.setTypedChars(copyTypedChars);
+      props.setText(props.text.slice(1));
+    } else if (incorrect) {
+      setWrongChar(true);
+    } else if (firstIncorrectThenIncorrect) {
+      const copyTypedChars = props.typedChars.slice();
+      copyTypedChars.push({
+        char: current,
+        status: "incorrect",
+      });
```
Comment (DevBastiS) [inline my-app/src/Components/Game.js:34]:
```suggestion
      });
      
```

**Review by DevBastiS — APPROVED** [review]
Just small stuff, remember that whitelines are your friends ;)

File: my-app/src/Components/TypingArea.js [hunk 4]
```diff
@@ -1,36 +1,55 @@
...
 import { TextRepository } from "../Repositories/TextRepository";
+import { TextLengthInput } from "./TextLengthInput";
 
 export function TypingArea() {
   const [text, setText] = useState("");
   const [textLength, setTextLength] = useState(5);
   const [textService, setTextService] = useState(new TextService(new TextRepository()));
+  const [typedChars, setTypedChars] = useState([]);
+  const [backupText, setBackupText] = useState("");
   useEffect(() => {
```
Comment (stefanluth) [inline my-app/src/Components/TypingArea.js]:
```suggestion
  const [backupText, setBackupText] = useState("");

  useEffect(() => {
```

File: my-app/src/Components/TypingArea.js [hunk 5]
```diff
@@ -1,36 +1,55 @@
...
+          onClick={() => {
+            setText(backupText);
+            setTypedChars([]);
+          }}
+        />
         <Button
           text="Generate New Text"
           onClick={() => {
-            textService.GenerateNewText(textLength).then((response) => setText(response));
+            newRun();
```
Comment (stefanluth) [inline my-app/src/Components/TypingArea.js]: You can remove the `{}`

**Review by stefanluth — APPROVED** [review]

**Pull request merged**